app/assets/javascripts/context_navigation.js

```diff
@@ -0,0 +1,384 @@
+class NavigationDocument {
+  constructor(el) {
+    this.el = $(el);
+  }
+
+  get id() {
+    return this.el.find('[data-document-id]').data().documentId;
+  }
+
+  setAsHighlighted() {
+    this.el.find('li.al-collection-context').addClass('al-hierarchy-highlight');
+  }
+
+  collapse() {
+    this.el.find('li.al-collection-context').addClass('collapsed');
+  }
+
+  render() {
+    return this.el.html();
+  }
+}
+
+/**
+ * Models the "Expand"/"Collapse" button, and provides an onClick event handler
+ * for the jQuery element
+ * @class
+ */
+class ExpandButton {
+  /**
+   * This retrieves the <li> elements which are hidden/rendered in response to
+   *   clicking the <button> element
+   * @param {jQuery} $li - the <button> element
+   * @return {jQuery} - a jQuery object containing the targeted <li>
+   */
+  findSiblings() {
+    const $siblings = this.$el.parent().children('li');
+    return $siblings.slice(0, -1);
+  }
+
+  /**
+   * This adds a "collapsed" class to all of the <li> children, as well as
+   *   updates the text for the <button> element
+   * @param {Event} event - the event propagated in response to clicking on the
+   *   <button> element
+   */
+  handleClick() {
+    const $targeted = this.findSiblings();
+
+    $targeted.toggleClass('collapsed');
+    this.$el.toggleClass('collapsed');
+
+    const containerText = this.$el.hasClass('collapsed') ? this.collapseText : this.expandText;
+    this.$el.text(containerText);
+  }
+
+  /**
+   * @constructor
+   */
+  constructor(data) {
+    this.collapseText = data.collapse;
+    this.expandText = data.expand;
+
+    this.$el = $(`<button class="my-3 btn btn-secondary btn-sm">${this.expandText}</button>`);
+    this.handleClick = this.handleClick.bind(this);
+    this.$el.click(this.handleClick);
+  }
+}
+
+/**
+ * Modeling <button> Elements which hide or retrieve <li> elements for sibling
+ *   documents nested within the <li> elements of the <ul> tree
+ * @class
+ */
+class NestedExpandButton extends ExpandButton {
+  /**
+   * This retrieves the <li> elements which are hidden/rendered in response to
+   *   clicking the <button> element
+   * @param {jQuery} $li - the <button> element
+   * @return {jQuery} - a jQuery object containing the targeted <li>
+   */
+  findSiblings() {
+    const highlighted = this.$el.siblings('.al-hierarchy-highlight');
+    const $siblings = highlighted.prevAll('.al-collection-context');
+    return $siblings.slice(0, -1);
+  }
+}
+
+/**
+ * Models the placeholder display elements for content loading from AJAX
+ *   requests
+ * @class
+ */
+class Placeholder {
+  /*
+   * Builds the element set which contains the placeholder markup
+   *   classes
+   */
+  /* eslint-disable class-methods-use-this */
+  buildElement() {
+    const elementMarkup = '<div class="al-hierarchy-placeholder">' +
+      '<h3 class="col-md-9"> <img alt="loading icon" src="/assets/ajax-loader.gif"> Loading...</h3>' +
```

Comment:
This is the bit that was changed

Reply:
Can you add a comment to the file where you made the change to add the loading icon instead of default gray lines.

Reply:
added comment